### PR TITLE
CI: Group Dependabot MRs

### PR DESCRIPTION
**Pull request by dolfinus**
### Problem

Currently Dependabot creates a new PR for each dependency it updates:
изображение

### Solution

My proposal is to create one PR for each dependency group, containing all the updates since previous week:
изображение

Here is an example:
https://github.com/MobileTeleSystems/horizon/blob/8b21b805254d1d0b118660f973d2f177e5368201/.github/dependabot.yml#L22-L26
https://github.com/MobileTeleSystems/horizon/pull/75
изображение

### Checklist

- [x] You've [signed-off](https://github.com/OpenLineage/OpenLineage/blob/main/why-the-dco.md) your work
- [X] Your pull request title follows our [guidelines](https://github.com/OpenLineage/OpenLineage/blob/main/CONTRIBUTING.md#creating-pull-requests)
- [ ] Your changes are accompanied by tests (_if relevant_)
- [ ] Your change contains a [small diff](https://kurtisnusbaum.medium.com/stacked-diffs-keeping-phabricator-diffs-small-d9964f4dcfa6) and is self-contained
- [ ] You've updated any relevant documentation (_if relevant_)
- [ ] Your comment includes a one-liner for the changelog about the specific purpose of the change (_not required for changes to tests, docs, or CI config_)
- [ ] You've versioned the core OpenLineage model or facets according to [SchemaVer](https://docs.snowplowanalytics.com/docs/pipeline-components-and-applications/iglu/common-architecture/schemaver) (_if relevant_)
- [ ] You've added a [header](https://github.com/OpenLineage/OpenLineage/tree/main/.github/header_templates.md) to source files (_if relevant_)

----
SPDX-License-Identifier: Apache-2.0\
Copyright 2018-2024 contributors to the OpenLineage project

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -32,6 +32,11 @@ updates:
     ignore:
       - dependency-name: "org.slf4j.slf4j-api"
       - dependency-name: "org.openapi.generator"
+    # https://til.simonwillison.net/github/dependabot-python-setup
+    groups:
+      python-packages:
+        patterns:
+          - '*'
```
Comment (kacpermuda) [inline .github/dependabot.yml:39]: I like the idea, but I rarely merge those updates so I'm not sure how often we do some selective merges, where we update some and packages and leave others. Let's hear from @mobuchowski or @pawel-big-lebowski here.

Also, shouldn't we at least split those PRs into multiple groups f.e. java / python / rust ? Or if they are to be grouped together, maybe we should not name the group `python-packages`?

Comment (dolfinus) [reply .github/dependabot.yml]: Fixed name

Comment (pawel-big-lebowski) [reply .github/dependabot.yml]: My only concern are versions we don't want to bump as CI is failing on new version. Like Hadoop map reduce on the screenshot attached from 2.10.2 -> 3.4.0. But I assume this should be solved telling dependabot to ignore certain packages.

Comment (dolfinus) [reply .github/dependabot.yml]: If this change introduces more issues than is solves, then I'm okay with closing this PR

Comment (pawel-big-lebowski) [reply .github/dependabot.yml]: I am happy to test this in action.

**Review by pawel-big-lebowski — APPROVED** [review]

File: .github/dependabot.yml (hunk 1 above)
Comment (pawel-big-lebowski) [reply .github/dependabot.yml]: Let's wait a day or two in case someone comes with strong `no`. Otherwise, we merge it.

**Review by mobuchowski — APPROVED** [review]

File: .github/dependabot.yml (hunk 1 above)
Comment (mobuchowski) [reply .github/dependabot.yml]: Very easy to reverse that change if we don't like - let's do it.

**Pull request merged**